hermeto/core/models/input.py

```diff
@@ -21,11 +31,33 @@
 ModelT = TypeVar("ModelT", bound=pydantic.BaseModel)
 
 
+def _is_experimental_package_error(error: "ErrorDict") -> Optional[tuple[str, str]]:
+    """Check if error is for experimental package and return (base_name, x_name) if so."""
+    if error.get("type") != "union_tag_invalid":
+        return None
+
+    tag = error.get("ctx", {}).get("tag", "")
+    x_prefixed = f"x-{tag}"
+
+    if x_prefixed in get_args(PackageManagerType):
+        return tag, x_prefixed
+
+    return None
+
+
 def parse_user_input(to_model: Callable[[T], ModelT], input_obj: T) -> ModelT:
     """Parse user input into a model, re-raise validation errors as InvalidInput."""
     try:
         return to_model(input_obj)
     except pydantic.ValidationError as e:
+        experimental_errors = [_is_experimental_package_error(err) for err in e.errors()]
+        experimental_suggestions = [exp for exp in experimental_errors if exp is not None]
```

Comment:
Non-blocking observation: you could have used the controversial `:=` operator here to make it into a one liner:
```python
experimental_pairs = [pair for err in e.errors() if (pair:=_is_experimental_package_error(err) is not None]
```
I have not decided yet whether this is an improvement or not, it does read a bit strange to me.